src/main/java/lotto/WinningStatistics.java

```diff
@@ -0,0 +1,28 @@
+package lotto;
+
+import java.util.Map;
+
+public class WinningStatistics {
+    private final Map<Rank, Integer> winningStatistics;
+
+    public WinningStatistics(final Map<Rank, Integer> winningStatistics) {
+        this.winningStatistics = winningStatistics;
+    }
+
+    public double calculateReturnRate(final int purchaseAmount) {
+        int totalWinningAmount = calculateTotalWinningAmount();
+        return (double) totalWinningAmount / purchaseAmount;
+    }
+
+    private int calculateTotalWinningAmount() {
+        int total = 0;
+        for (final Rank rank : winningStatistics.keySet()) {
+            total += rank.calculateAllWinningAmount(winningStatistics.get(rank));
+        }
+        return total;
```

Comment:
아래와 같이 변경해볼 수도 있을 것 같아요. 😉 

```suggestion
        return winningStatistics.entrySet()
            .stream()
            .mapToInt(entry -> entry.getKey().calculateAllWinningAmount(entry.getValue()))
            .sum();
```